File: back/api/schemas/breed.py
# -*- coding: utf-8 -*-
"""
schemas/breed.py
----------------
견종 도메인 Pydantic v2 스키마.

BreedResponse: 견종 단건/목록 응답 스키마
"""

from __future__ import annotations

from datetime import datetime

from enum import Enum
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class BreedSizeEnum(str, Enum):
    small = "small"
    medium = "medium"
    large = "large"

class ActivityLevelEnum(str, Enum):
    low = "low"
    medium = "medium"
    high = "high"


class BreedResponse(BaseModel):
    """견종 단건 응답."""

    model_config = ConfigDict(from_attributes=True)

    id: int = Field(..., description="견종 ID")
    name_ko: str = Field(..., description="견종명 (한국어)")
    name_en: str = Field(..., description="견종명 (영어)")
    top10: bool = Field(..., description="인기 TOP10 여부")
    size: Optional[BreedSizeEnum] = Field(None, description="견종 크기 분류")
    activity_level: Optional[ActivityLevelEnum] = Field(None, description="활동량 수준")
    temperament: Optional[str] = Field(None, description="성격 태그")
    breed_group: Optional[str] = Field(None, description="견종 그룹")
    image_url: Optional[str] = Field(None, description="견종 대표 이미지 URL")
    created_at: datetime = Field(..., description="등록 일시")

File: mobile/lib/shared/models/breed.dart
import 'enums.dart';

/// 백엔드 `schemas/breed.py BreedResponse` 와 1:1.
class Breed {
  const Breed({
    required this.id,
    required this.nameKo,
    required this.nameEn,
    required this.top10,
    this.size,
    this.activityLevel,
    this.temperament,
    this.breedGroup,
    this.imageUrl,
  });

  final int id;
  final String nameKo;
  final String nameEn;
  final bool top10;
  final BreedSize? size;
  final String? activityLevel;
  final String? temperament;
  final String? breedGroup;
  final String? imageUrl;

  factory Breed.fromJson(Map<String, dynamic> json) {
    return Breed(
      id: json['id'] as int,
      nameKo: json['name_ko'] as String,
      nameEn: json['name_en'] as String,
      top10: json['top10'] as bool,
      size: BreedSize.fromWire(json['size'] as String?),
      activityLevel: json['activity_level'] as String?,
      temperament: json['temperament'] as String?,
      breedGroup: json['breed_group'] as String?,
      imageUrl: json['image_url'] as String?,
    );
  }
}
